lib/net.js

```diff
@@ -1018,21 +1020,25 @@ function lookupAndConnect(self, options) {
   var localPort = options.localPort;
 
   if (localAddress && !cares.isIP(localAddress)) {
-    throw new TypeError('"localAddress" option must be a valid IP: ' +
-                        localAddress);
+    throw new errors.TypeError('ERR_INVALID_IP_ADDRESS', localAddress);
   }
 
   if (localPort && typeof localPort !== 'number') {
-    throw new TypeError('"localPort" option should be a number: ' + localPort);
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE',
+                               'options.localPort',
+                               'number',
+                               localPort);
   }
 
   if (typeof port !== 'undefined') {
     if (typeof port !== 'number' && typeof port !== 'string') {
-      throw new TypeError('"port" option should be a number or string: ' +
-                          port);
+      throw new errors.TypeError('ERR_INVALID_ARG_TYPE',
+                                 'options.port',
+                                 ['number', 'string'],
+                                 port);
     }
     if (!isLegalPort(port)) {
-      throw new RangeError('"port" option should be >= 0 and < 65536: ' + port);
+      throw new errors.RangeError('ERR_SOCKET_BAD_PORT', port);
```

Comment:
The error type must also be able to handle the provided port.

Reply:
I have made a change to include the port in the err msg if the port arg is passed to the error constructor.